tokio/src/sync/semaphore.rs

```diff
@@ -123,7 +123,14 @@ fn bounds() {
 }
 
 impl Semaphore {
+    /// The maximum number of permits which a semaphore can hold.
+    ///
+    /// Exceeding this limit typically results in a panic.
+    pub const MAX_PERMITS: usize = super::batch_semaphore::Semaphore::MAX_PERMITS;
+
     /// Creates a new semaphore with the initial number of permits.
+    ///
+    /// Panics if `permits` exceeds [`Semaphore::MAX_PERMITS`].
     #[track_caller]
     pub fn new(permits: usize) -> Self {
```

Comment:
Can you please add a test that verifies that it does _not_ panic if you pass `MAX_PERMITS`, and that it does panic if you pass `MAX_PERMITS + 1`.

Reply:
Added simple tests.

Or do you mean complex tests that actually check `Semaphore` behaviour when it is near MAX_PERMITS?

Reply:
We already have tests for the semaphore. I just wanted a simple test for the constant.